Automattic/sensei
# Update some block patterns and remove others

Resolves pattern issues documented in p6rkRX-6Ia-p2.

## Proposed Changes
When we first built course patterns, we built them against designs that were optimized for the Course theme. As such, the styles were much too opinionated. It's become more important for us to ensure our patterns look decent across most themes, especially block themes.

To get our patterns into a good state, this PR removes a couple of them (the list-based Course List block ones) and fixes up the Landing Page and Course List patterns. I've kept the old versions of the Landing Pages, but moved them to a `v1` folder. The new pattern is in the `v2` folder. This means it won't be possible to compare the before and after for this file. ☹️ 

The only difference between the Landing Page pattern vs. the Landing Page - Grid pattern was the layout of the Course List block (either list or grid). Given https://github.com/Automattic/sensei/issues/7266, and how tricky I think it will be to resolve, I've made a decision to remove both of these patterns. We have more pressing things to work on, so I've opted to remove these patterns instead of leaving them in a broken state. I've renamed the Landing Page - Grid and Course List - Grid patterns to just Landing Page and Course List.

I tried to put as much of the styling into the new pattern (vs. using CSS) as possible. However, in some cases I left some CSS so as not to break pages that are using the now-stale patterns. My focus was on the new patterns though, not on fixing pages using the old pattern. Those sites will be able to make their pages look better by switching to the new pattern. Even so, the old pages will still benefit from these changes, especially the landing page header text and testimonial, and especially on Divi and Astra.

The new Landing Page and Course List patterns, though still optimized for Course, use a utilitarian approach (p6rkRX-6Ia-p2#comment-6771) and should now look reasonable on most block themes. It will not be possible to make it look good on _all_ block themes though (e.g. TT3), since different themes use different naming for presets and other differences. And although the new patterns will look better on Divi and Astra, Divi in particular still has issues that I didn't try to resolve because classic themes aren't the focus of this work (and because Divi makes it so hard to make things look good).

## Testing Instructions


1. On `trunk`, create a page that uses the Landing Page - Grid pattern and another page that uses the Course List - Grid pattern (we'll use these to compare before and after).
2. Open these pages in new tabs using a few different themes.
3. Switch to this branch and `npm run build`.
4. Open the pages you created previously in different tabs for the same themes and compare the before and after. Depending on the themes you use, the after may look slightly better, but it shouldn't look any worse.
5. Create new pages that use the Landing Page and Course List patterns.
6. Again, open these pages on the same themes in different tabs. They should look a lot better, especially on Annalee (pNEWy-hkA-p2) and TT2.
7. Check mobile view as well to ensure those look decent.

## Screenshots
Partial screenshots of the top part of the Landing Page pattern:

### Astra - Before
![Astra - Before](https://github.com/Automattic/sensei/assets/1190420/a4962cc8-7966-48f1-8679-66ce8bfe2012)

### Astra - After
![Astra - After](https://github.com/Automattic/sensei/assets/1190420/347da053-dcfd-46e0-86e3-cfda8f873278)

### Annalee - Before
![Annalee - Before](https://github.com/Automattic/sensei/assets/1190420/8b474d4a-3eda-4f06-a8e0-a70d6df90257)

### Annalee - After
![Annalee - After](https://github.com/Automattic/sensei/assets/1190420/ad9d978e-b2c4-41cd-a178-40d947fe50f5)

### TT2 - Before
![TT2 - Before](https://github.com/Automattic/sensei/assets/1190420/64781644-6900-4847-9c18-faff57865805)

### TT2 - After
![TT2 - After](https://github.com/Automattic/sensei/assets/1190420/13f9daee-de38-40e3-95dc-dbe3a965c15d)

## Pre-Merge Checklist
